### PR TITLE
Improve ising gates documentation

### PR DESCRIPTION
**Pull request by rmoyard**
**Description of the Change:**

This PR improves the documentation for Ising gates.

### CODE REVIEW

File: pennylane/ops/qubit/parametric_ops.py [hunk 1]
```diff
@@ -2306,13 +2306,18 @@ class IsingXX(Operation):
...
             \cos(\phi / 2) & 0 & 0 & -i \sin(\phi / 2) \\
             0 & \cos(\phi / 2) & -i \sin(\phi / 2) & 0 \\
             0 & -i \sin(\phi / 2) & \cos(\phi / 2) & 0 \\
             -i \sin(\phi / 2) & 0 & 0 & \cos(\phi / 2)
         \end{bmatrix}.
 
+    .. note::
+
+        :math:`XX(0) = I, XX(\pi) = i (X \otimes X)`
+
```
Comment (antalszava) [inline pennylane/ops/qubit/parametric_ops.py:2320]: Adding a bit of text and itemising the equation for each operation may be worth it (the example is just a suggestion).
```suggestion

        Special cases of using the :math:`XX` operator include:
        
        * :math:`XX(0) = I`;
        * :math:`XX(\pi) = i (X \otimes X)`.

```

**Review by antalszava — APPROVED** [review]
:partying_face: Looks great :clap:

File: pennylane/ops/qubit/parametric_ops.py [hunk 2]
```diff
@@ -2429,13 +2437,18 @@ class IsingYY(Operation):
...
+        \begin{bmatrix}
+            \cos(\phi / 2) & 0 & 0 & i \sin(\phi / 2) \\
+            0 & \cos(\phi / 2) & -i \sin(\phi / 2) & 0 \\
+            0 & -i \sin(\phi / 2) & \cos(\phi / 2) & 0 \\
+            i \sin(\phi / 2) & 0 & 0 & \cos(\phi / 2)
         \end{bmatrix}.
 
+    .. note::
+
+        :math:`YY(0) = I, YY(\pi) = i (Y \otimes Y)`
```
Comment (rmoyard) [inline pennylane/ops/qubit/parametric_ops.py:2450]:
```suggestion
        Special cases of using the :math:`YY` operator include:
        
        * :math:`YY(0) = I`;
        * :math:`YY(\pi) = i (Y \otimes Y)`.
```

File: pennylane/ops/qubit/parametric_ops.py [hunk 3]
```diff
@@ -2551,13 +2564,18 @@ class IsingZZ(Operation):
...
+        \begin{bmatrix}
+            e^{-i \phi / 2} & 0 & 0 & 0 \\
+            0 & e^{i \phi / 2} & 0 & 0 \\
+            0 & 0 & e^{i \phi / 2} & 0 \\
+            0 & 0 & 0 & e^{-i \phi / 2}
         \end{bmatrix}.
 
+    .. note::
+
+        :math:`ZZ(0) = I, ZZ(\pi) = - (Z \otimes Z), ZZ(2\pi) = - I`
```
Comment (rmoyard) [inline pennylane/ops/qubit/parametric_ops.py:2577]:
```suggestion
        Special cases of using the :math:`ZZ` operator include:
        
        * :math:`ZZ(0) = I`;
        * :math:`ZZ(\pi) = - (Z \otimes Z)`;
        * :math:`ZZ(2\pi) = - I;
```

File: pennylane/ops/qubit/parametric_ops.py [hunk 4]
```diff
@@ -2702,15 +2720,20 @@ def pow(self, z):
...
+        \begin{bmatrix}
             1 & 0 & 0 & 0 \\
             0 & \cos(\phi / 2) & i \sin(\phi / 2) & 0 \\
             0 & i \sin(\phi / 2) & \cos(\phi / 2) & 0 \\
             0 & 0 & 0 & 1
         \end{bmatrix}.
 
+    .. note::
+
+        :math:`XY(0) = I, XY(\frac{\pi}{2}) = \sqrt{iSWAP}, XY(\pi) = iSWAP`
```
Comment (rmoyard) [inline pennylane/ops/qubit/parametric_ops.py:2735]:
```suggestion
        Special cases of using the :math:`XY` operator include:
        
        * :math:`XY(0) = I`;
        * :math:`XY(\frac{\pi}{2}) = \sqrt{iSWAP}`;
        * :math:`XY(\pi) = iSWAP;
```

**Pull request merged**